beetsplug/edit.py

```diff
@@ -243,9 +244,9 @@ def edit_objects(self, objs, fields):
         old_data = [flatten(o, fields) for o in objs]
 
         # Set up a temporary file with the initial data for editing.
-        new = NamedTemporaryFile(mode='w', suffix='.yaml', delete=False)
+        new = NamedTemporaryFile(suffix='.yaml', delete=False)
         old_str = dump(old_data)
-        new.write(old_str)
+        new.write(old_str.encode('utf-8'))
```

Comment:
As I understand it, `yaml.dump` is inconsistent across Python versions? That is, on Python 2, it produces UTF-8-encoded bytes, and on Python 3, it produces unicode. Seems like a strange decision, but OK I guess.

Because the `NamedTemporaryFile` call appears to open the file in binary mode on both versions, you're right to want to produce bytes here. I think we just need something like:

```py
old_str if PY2 else old_str.encode('utf8')
```

to get bytes on both versions.